### ISSUE TITLE
FileBlobProvider does not respect configured path

### ISSUE THREAD

**Issue by tholee** [episerver/netcore-preview#99]
Adding the configuration below does not have the expected effect in the Alloy netcore-preview project.
The default "[appDataPath]\blobs" is still used and not my configured: "[appDataPath]\\my-custom-blobs-folder"
```json
"AllowedHosts": "*",
  "EPiServer": {
    "Cms": {
      "BlobProviders": {
        "DefaultProvider": "FileBlobProvider",
        "Providers": {
          "FileBlobProvider": "EPiServer.Framework.Blobs.FileBlobProvider, EPiServer.Framework"
        }
      },
      "FileBlobProvider": {
        "Path": "[appDataPath]\\my-custom-blobs-folder"
```
The same goes for configuration from code.
```c#
services.Configure<FileBlobProviderOptions>(o => 
{
    o.Path = @"[appDataPath]\my-custom-blobs-folder";
});
```
It seems like the behavior is due to constructor creating a new option instance rather that using dependency injection
```c#
public FileBlobProvider(IPhysicalPathResolver physicalPathResolver) 
       : this(physicalPathResolver, new FileBlobProviderOptions())
{
}
```

A workaround could be
```c#
public class MyWorkaroundFileBlobProvider : FileBlobProvider
{
    public MyWorkaroundFileBlobProvider (IPhysicalPathResolver physicalPathResolver, IOptions<FileBlobProviderOptions> options)
        : base(physicalPathResolver, options.Value)
    {
    }
}
```

**Comment by lunchin** [comment]
Thanks for reporting the issue, I have added to our internal issue tracker.

**Comment by bjuris** [comment]
Fixed in next drop.

An easy workaround to make sure it uses the correct constructor is registering it again **after** services.AddCms():
```csharp
services.AddTransient<EPiServer.Framework.Blobs.FileBlobProvider>();
```
and then you can configure:
```csharp
services.Configure<FileBlobProviderOptions>(o => 
{
    o.Path = @"[appDataPath]\my-custom-blobs-folder";
});
```

**Comment by lunchin** [comment]
Fixed 12.0.1
